common/scala/src/main/scala/whisk/common/tracing/TraceUtil.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common.tracing
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.collection.mutable.Map
```

Comment:
Per comments below: Don't import mutable datastructures, but rather import `scala.collection.mutable` and use the explicit type, like `mutable.Map` to not confuse mutable and immutable structures.